containers/jetty/src/main/java/com/yahoo/athenz/container/log/JettyConnectionLogger/JettyConnectionLogger.java

```diff
@@ -0,0 +1,284 @@
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package com.yahoo.athenz.container.log.JettyConnectionLogger;
+
+import com.yahoo.athenz.common.metrics.Metric;
+import org.eclipse.jetty.io.Connection;
+import org.eclipse.jetty.io.EndPoint;
+import org.eclipse.jetty.io.SocketChannelEndPoint;
+import org.eclipse.jetty.io.ssl.SslConnection;
+import org.eclipse.jetty.io.ssl.SslHandshakeListener;
+import org.eclipse.jetty.server.*;
+import org.eclipse.jetty.util.component.AbstractLifeCycle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLHandshakeException;
+import java.net.InetSocketAddress;
+import java.time.Instant;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.yahoo.athenz.container.log.JettyConnectionLogger.ExceptionCauseFetcher.getInnerCause;
+
+public class JettyConnectionLogger extends AbstractLifeCycle implements Connection.Listener, HttpChannel.Listener, SslHandshakeListener {
+
+    public static final String GENERAL_SSL_ERROR = "General SSLEngine problem";
+    public static final String METRIC_NAME = "ssl_error";
+    private static final Logger LOGGER = LoggerFactory.getLogger(JettyConnectionLogger.class.getName());
+
+    private final ConcurrentMap<SocketChannelEndPoint, ConnectionInfo> connectionInfo = new ConcurrentHashMap<>();
+    private final ConcurrentMap<SSLEngine, ConnectionInfo> sslToConnectionInfo = new ConcurrentHashMap<>();
+
+    private final ConnectionLog connectionLog;
+    private final Metric metric;
+
+    public JettyConnectionLogger(ConnectionLog connectionLog, Metric metric) {
+        this.connectionLog = connectionLog;
+        this.metric = metric;
+        LOGGER.info("Jetty connection logger is running");
+    }
+
+    //
+    // AbstractLifeCycle methods start
+    //
+    @Override
+    protected void doStop() {
+        handleListenerInvocation("AbstractLifeCycle", "doStop", "", Collections.emptyList(), () -> {
+            LOGGER.info("Jetty connection logger is stopped");
+        });
+    }
+
+    @Override
+    protected void doStart() {
+        handleListenerInvocation("AbstractLifeCycle", "doStart", "", Collections.emptyList(), () -> {
+            LOGGER.info("Jetty connection logger is started");
+        });
+    }
+    //
+    // AbstractLifeCycle methods stop
+    //
+
+    //
+    // Connection.Listener methods start
+    //
+    @Override
+    public void onOpened(Connection connection) {
+        handleListenerInvocation("Connection.Listener", "onOpened", "%h", Stream.of(connection).collect(Collectors.toList()), () -> {
+            SocketChannelEndPoint endpoint = findUnderlyingSocketEndpoint(connection.getEndPoint());
```

Comment:
When a connection is opened, get the connection details and put them in the "ConnectionInfo" map. They will be removed from the map when the connection closes. The details of the connection will only be written to log in case of handshake failure (handshakeFailed was called with the exception)